sdk-base/src/main/java/com/mapbox/maps/extension/observable/ObservableExtension.kt

```diff
@@ -301,124 +299,329 @@ fun ObservableInterface.unsubscribeSourceRemoved(observer: Observer) =
  */
+@Suppress("UNCHECKED_CAST")
 fun Event.getStyleImageMissingEventData(): StyleImageMissingEventData {
-  val json = data.toJson()
-  return gson.fromJson(json, StyleImageMissingEventData::class.java)
+  val map = data.contents as Map<String, Value>
+  return StyleImageMissingEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END),
+    id = map.nonNullString(ID)
+  )
 }
 
 /**
  * Get the parsed event data for style image unused event.
  * @return a parsed StyleImageUnusedEventData.
  */
+@Suppress("UNCHECKED_CAST")
 fun Event.getStyleImageUnusedEventData(): StyleImageUnusedEventData {
-  val json = data.toJson()
-  return gson.fromJson(json, StyleImageUnusedEventData::class.java)
+  val map = data.contents as Map<String, Value>
+  return StyleImageUnusedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END),
+    id = map.nonNullString(ID)
+  )
 }
 
 /**
  * Get the parsed event data for source added event.
  * @return a parsed IDStringEventData.
  */
+@Suppress("UNCHECKED_CAST")
 fun Event.getSourceAddedEventData(): SourceAddedEventData {
-  val json = data.toJson()
-  return gson.fromJson(json, SourceAddedEventData::class.java)
+  val map = data.contents as Map<String, Value>
+  return SourceAddedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END),
+    id = map.nonNullString(ID)
+  )
 }
 
 /**
  * Get the parsed event data for source removed event.
  * @return a parsed SourceRemovedEventData.
  */
+@Suppress("UNCHECKED_CAST")
 fun Event.getSourceRemovedEventData(): SourceRemovedEventData {
-  val json = data.toJson()
-  return gson.fromJson(json, SourceRemovedEventData::class.java)
+  val map = data.contents as Map<String, Value>
+  return SourceRemovedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END),
+    id = map.nonNullString(ID)
+  )
 }
 
 /**
  * Get the parsed event data for render frame started event.
  * @return a parsed RenderFrameStartedEventData.
  */
+@Suppress("UNCHECKED_CAST")
 fun Event.getRenderFrameStartedEventData(): RenderFrameStartedEventData {
-  val json = data.toJson()
-  return gson.fromJson(json, RenderFrameStartedEventData::class.java)
+  val map = data.contents as Map<String, Value>
+  return RenderFrameStartedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END)
+  )
 }
 
 /**
  * Get the parsed event data for render frame finished event.
  * @return a parsed RenderFrameFinishedEventData.
  */
+@Suppress("UNCHECKED_CAST")
 fun Event.getRenderFrameFinishedEventData(): RenderFrameFinishedEventData {
-  val json = data.toJson()
-  return gson.fromJson(json, RenderFrameFinishedEventData::class.java)
+  val map = data.contents as Map<String, Value>
+  return RenderFrameFinishedEventData(
+    begin = map.nonNullLong(BEGIN),
+    end = map.nullableLong(END),
+    renderMode = RenderMode.valueOf(map.nonNullString(RENDER_MODE).toUpperCase(Locale.US)),
```

Comment:
how about one more extension
```suggestion
    renderMode = RenderMode.valueOf(map.nonNullStringWithUpperCase(RENDER_MODE)),
```

```
fun Map<>.nonNullStringWithUpperCase(name: String, locale: Locale = Locale.US) =  
this.nonNullString(name).toUpperCase(locale)
```

Reply:
Looks good! Added this function